### PR TITLE
toolchain.mk: add clang-toolchains target

### PR DESCRIPTION
**Pull request by jforissier**
Add "make clang-toolchains" to download and extract Clang 9.0.1 into
$(ROOT)/clang-9.0.1. Usage:

 $ make clang-toolchains
 $ PATH=../clang-9.0.1/bin:$PATH make COMPILER=clang

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

File: get_clang.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+function cleanup() {
+  rm -f ${X86_64}.tar.xz ${AARCH64}.tar.xz ${ARMV7A}.tar.xz
+  rm -rf ${AARCH64} ${ARMV7A}
+}
+
+trap "{ exit 2; }" INT
+trap cleanup EXIT
+
+set -x
```
Comment (vchong) [inline get_clang.sh:38]: Copy paste typo or do you mean to `set +x` instead?

Comment (jforissier) [reply get_clang.sh:38]: I really meant `set -x` in order to show progress and help diagnose any issue. The output is not very tidy but I thought it is a simple and effective solution.

Comment (vchong) [reply get_clang.sh:38]: Isn't it already set in L21?

Comment (jforissier) [reply get_clang.sh:38]: Huh, yes. I was wondering why you were mentioning copy/paste :confused: Will remove this one, thanks.

**Comment by jforissier** [comment]
> `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with suggestion in commit log.
> 
> To avoid path issue in the build process, suggestion to change commit log:
> 
> ```diff
> - $ PATH=../clang-9.0.1/bin:$PATH make COMPILER=clang
> + $ PATH=$(PWD)/../clang-9.0.1/bin:$PATH make COMPILER=clang
> ```

Will fix, thanks.

**Comment by etienne-lms** [comment]
> Will fix, thanks. It is $PWD or ${PWD} in the shell by the way.

indeed :)

**Pull request merged**